### PR TITLE
Cleanup MapUserSessionAdapter.getAuthenticatedClientSessions()

### LINKED ISSUE #14743: Observations after reviewing MapUserSessionAdapter.getAuthenticatedClientSessions()
### Description

When reviewing MapUserSessionAdapter.getAuthenticatedClientSessions(), I found some smaller issues where too many entities were fetched and a method name was misleading. 

Also the map store will only return entities that are still valid and not expired (this changed recently, this is probably old code).

This issue will fix those. 

### Motivation

Removing unnecessary code, clear method names

### PR DESCRIPTION
**Pull request by ahus1**
Closes #14743

### CODE REVIEW

**Comment by ahus1** [comment]
Those changes are supposed to be small and local. If they lead t a lot of questions, I'd rather revert them.

File: model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProvider.java [hunk 1]
```diff
@@ -80,16 +80,14 @@ private Function<MapUserSessionEntity, UserSessionModel> userEntityToAdapterFunc
         // Clone entity before returning back, to avoid giving away a reference to the live object to the caller
         return (origEntity) -> {
             if (origEntity == null) return null;
-            if (isExpired(origEntity, false)) {
-                if (TRANSIENT == origEntity.getPersistenceState()) {
+            if (TRANSIENT == origEntity.getPersistenceState()) {
+                // only check expiry for transient entities, as the physical map store will not return expired entities on any query
```
Comment (hmlnarik) [inline model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProvider.java:84]: Unfortunately, this assumption is not correct. The expiration check for persistent entities has to remain here, see the following javadoc.
https://github.com/keycloak/keycloak/blob/148c7695ff5b32f46f05a5f341ca5456da144987/model/map/src/main/java/org/keycloak/models/map/common/ExpirableEntity.java#L24-L27

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you for the PR! Up to the changes in the `MapUserSessionProvider` looks good to me, please see inline.

**Review by hmlnarik — APPROVED** [review]
Approving (conditioned on green builds). Thank you for the PR!

**Review by sguilhen — COMMENTED** [review]
Operator CI seems to be failing not only on this PR. Got the same error in the remote tests in a completely unrelated PR

**Comment by ahus1** [comment]
Builds are green, operator failure is not related as I see it - thank for Stefan confirming it.

**Review by mhajas — APPROVED** [review]
Thanks @ahus1, LGTM

**Pull request merged**